app/services/requirements/file_attachment_upload_checker.rb

```diff
@@ -63,8 +70,23 @@ def issues
   private
 
     def unsupported_type?
-      type = Marcel::MimeType.for(file, name: file.original_filename)
-      ALLOWED_FORMATS.exclude? type
+      (ALLOWED_FORMATS + [ALLOWED_ZIP]).exclude? content_type
+    end
+
+    def invalid_zip?
+      return if content_type != ALLOWED_ZIP
+
+      Zip::File.open(file) do |zf|
+        zf.each do |f|
```

Comment:
Is zf an enumerable? If so I'd suggest using a method like all? or any? rather than an each with a return inside it (my instinct when seeing that is thinking you meant a next)

Reply:
Yep good shout, didn't think to use the methods you suggested! :)